debug/gdbserver.py

```diff
@@ -728,14 +760,26 @@ def test(self):
         self.gdb.c()
         before = self.gdb.p("$pc")
         assertEqual(before, self.gdb.p("&crc32a"))
+
         self.gdb.stepi()
         after = self.gdb.p("$pc")
-        assertNotEqual(before, after)
+        if self.reserve_trigger_supported:
+            # OpenOCD should not disable reserved triggers by itself.
+            assertEqual(before, after)
+        else:
+            assertNotEqual(before, after)
 
         # Remove the manual HW breakpoint.
         assertEqual(tselect, self.gdb.p("$tselect"))
         self.gdb.p("$tdata1=0")
 
+        if self.reserve_trigger_supported:
+            self.gdb.stepi()
+            after = self.gdb.p("$pc")
+            assertNotEqual(before, after)
```

Comment:
please, consider a short annotation. for these asserts where you think we may provide some useful context.

Reply:
Addressed.